### PR TITLE
test

### PR DESCRIPTION
**Pull request by vkucher49**

### CODE REVIEW

File: src/main/java/mate/academy/dto/category/CategoryRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package mate.academy.dto.category;
+
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+@Data
+public class CategoryRequestDto {
+    @NotNull
+    private String name;
+    private String description;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/dto/category/CategoryRequestDto.java]: Valid description, is is possible empty fields?

File: src/main/java/mate/academy/model/Book.java [hunk 2]
```diff
@@ -2,31 +2,52 @@
...
+    @Column(name = "cover_image")
+    private String coverImage;
+    @ManyToMany(fetch = FetchType.EAGER)
+    @JoinTable(name = "books_categories",
+            joinColumns = @JoinColumn(name = "book_id"),
+            inverseJoinColumns = @JoinColumn(name = "category_id"))
+    private Set<Category> categories = new HashSet<>();
+    @ManyToOne
+    @JoinColumn(name = "category_id", nullable = false)
+    private Category category;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/model/Book.java]: You have set of categories and category field?

File: src/main/java/mate/academy/model/Category.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
+    private String name;
+    private String description;
+    @Column(nullable = false, name = "is_deleted")
+    private boolean isDeleted = false;
+    @OneToMany(mappedBy = "category")
+    private List<Book> books;
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/model/Category.java]: Are you sure?

File: src/main/java/mate/academy/model/Role.java [hunk 4]
```diff
@@ -21,7 +25,7 @@ public class Role {
     @Enumerated(value = EnumType.STRING)
     private RoleName roleName;
     @Column(name = "is_deleted")
-    private boolean isDeleted;
+    private boolean isDeleted = false;
 
     public enum RoleName {
         ROLE_USER, ROLE_ADMIN
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/model/Role.java]: Write enums in stack

File: src/main/java/mate/academy/service/CategoryServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        Category categoryFromDb = categoryRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find category by id = " + id)
+        );
+        return categoryMapper.toDto(categoryFromDb);
+    }
+
+    @Override
+    public CategoryResponseDto save(CategoryRequestDto categoryDto) {
+        Category category = categoryMapper.toEntity(categoryDto);
+        Category saveCategory = categoryRepository.save(category);
```
Comment (Sanderfuck) [inline src/main/java/mate/academy/service/CategoryServiceImpl.java]:
```suggestion
        Category savedCategory = categoryRepository.save(category);
```

**Review by Sanderfuck — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/controller/CategoryController.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    @DeleteMapping("/{id}")
+    @Operation(summary = "Delete category by id", description = "Delete category by id from db")
+    public void delete(@PathVariable Long id) {
+        categoryService.deleteById(id);
+    }
+
+    @GetMapping("/books/{id}")
+    @Operation(summary = "Get books by category id", description = "Get books from db by category")
+    public List<BookDtoWithoutCategoryIds> getBooksByCategoryId(@PathVariable Long id) {
```
Comment (Rommelua) [inline src/main/java/mate/academy/controller/CategoryController.java]: image

**Pull request merged**